### PR TITLE
Feature/43537 ps report data sync answers

### PR DESCRIPTION
**Pull request by jon-shipley**
Move answers and check score table to mtc_results table

* Change sync function to be event driven
* Remove old code that required picking up data from Azure table storage
* Add new 'functions-app' SQL user and perms
* Rename migration

A later PR will move function to the throttled functions.

### CODE REVIEW

File: functions-app/sync-results-to-sql/service/data-access/data.service.js [hunk 1]
```diff
@@ -5,131 +5,25 @@ const R = require('ramda')
...
-                    AND cs.id IS NULL`
-    return sqlService.query(sql)
-  },
-
-  /**
-   * Return an indexed object of all the questions 1x1 to 12x12 - includes warmup questions which are distinct
+   * Return an indexed object of all the live questions 1x1 to 12x12 - includes warmup questions which are distinct
    * Data is cached once retrieved
-   * @return {Promise<{object}>}  { '1x1': {id: 145, factor1: 1, factor2: 1, isWarmUp: false, code: 'Q001' }, ... }
+   * @return {Promise<Readonly<{}>}  { '1x1': {id: 145, factor1: 1, factor2: 1, isWarmUp: false, code: 'Q001' }, ... }
```
Comment (GuyHarwood) [inline functions-app/sync-results-to-sql/service/data-access/data.service.js:13]: nice 💯

File: functions-app/sync-results-to-sql/service/results.service.js [hunk 2]
```diff
@@ -1,103 +1,61 @@
...
 
-  /**
-   * Get schools that have new checks to process
-   * Defined as having completed checks in the DB, but where the marking data (checkScores table) is missing.
-   * @return {Promise<{id: number, name: string, schoolGuid: string}>}
-   */
-  getSchoolsWithNewChecks: async function getSchoolsWithNewChecks () {
-    return dataService.sqlGetSchoolsWithNewChecks()
+  prepareCheckResult: function prepareCheckResult (markedCheck) {
+    console.log('prepareCheckResult: ', markedCheck)
```
Comment (GuyHarwood) [inline functions-app/sync-results-to-sql/service/results.service.js:15]: intentional debug statement?

File: tslib/src/functions/check-marker/check-marker.v1.ts [hunk 3]
```diff
@@ -24,20 +28,33 @@ export class CheckMarkerV1 {
     } else {
       this.sqlService = sqlService
     }
+
+    if (sqlService === undefined) {
+      this.compressionService = new CompressionService()
+    } else {
+      this.compressionService = new CompressionService()
```
Comment (GuyHarwood) [inline tslib/src/functions/check-marker/check-marker.v1.ts:35]: intentional?

**Review by GuyHarwood — CHANGES_REQUESTED** [review]
LGSF, just a couple of things to address in comments

File: functions-app/sync-results-to-sql/service/results.service.js (hunk 2 above)
Comment (jon-shipley) [reply functions-app/sync-results-to-sql/service/results.service.js]: No 😊 , removed

File: tslib/src/functions/check-marker/check-marker.v1.ts (hunk 3 above)
Comment (jon-shipley) [reply tslib/src/functions/check-marker/check-marker.v1.ts]: oops 😊 , now fixed

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**